libraries/botbuilder-dialogs/src/dialogContext.ts

```diff
@@ -191,14 +191,20 @@ export class DialogContext {
      * Obtain the CultureInfo in DialogContext.
      * @returns a locale string.
      */
-    public getLocale(): string
-    {
-        let locale: string = undefined;
-        if (this.context.locale !== undefined && this.context.locale !== null) {
-            locale = this.context.locale;
+    public getLocale(): string {
+        const _turnLocaleProperty = 'turn.locale';
+
+        const _trunLocaleValue = this.state.getValue(_turnLocaleProperty);
+        if (_trunLocaleValue !== undefined && _trunLocaleValue !== '') {
```

Comment:
A simple truthy check would handle both of these cases.

```typescript
if (turnLocaleValue) {
  return turnLocaleValue;
}
```

Reply:
changed